### PR TITLE
fix(ci): Do not skip step for visual diffs on PRs

### PR DESCRIPTION
**Pull request by armenzg**
This fixes a regression from #33609 since `head_repository.full_name` is not part of the `github` context in PRs.

Visual Snapshots have not been running from acceptance/visual-diff for the last week. In the PR,
we missed that it did not get triggered:
https://github.com/getsentry/sentry/actions/runs/2167609568

The step (rather than the job) is being skipped on PRs:
https://github.com/getsentry/sentry/runs/6117835780?check_suite_focus=true

This has not been noticed because `visual-diff.yml` would still trigger Visual Snapshots for PRs:
https://github.com/getsentry/sentry/pull/33835/checks?check_run_id=6117841569

### CODE REVIEW

File: .github/workflows/visual-diff.yml [hunk 1]
```diff
@@ -12,8 +12,8 @@ on:
 
 jobs:
   visual-diff:
-    # This is the opposite condition as acceptance.yml/visual-diff
-    if: github.head_repository.full_name != 'getsentry/sentry'
+    # Only execute this check when a PR is opened from a fork rather than the upstream repo & master pushes
+    if: github.event.workflow_run.head_repository.full_name != 'getsentry/sentry' || github.event.workflow_run.head_branch == 'master'
```
Comment (armenzg) [inline .github/workflows/visual-diff.yml:16]: Ideally, `master` pushes would execute `visual-diff` from the acceptance workflow, however, as I found out in PR #33964 we can't at the moment.

This screenshot shows that a push to `master` would trigger the workflow run while a push to a PR would not:
image

Comment (armenzg) [reply .github/workflows/visual-diff.yml:16]: This also shows that a check on a PR for a fork would execute (unlike the PR from within the upstream repo):
image

**Review by armenzg — COMMENTED** [review]
I will not be landing this until early in a morning.

File: .github/workflows/acceptance.yml [hunk 2]
```diff
@@ -324,8 +324,8 @@ jobs:
       - name: Diff snapshots
         id: visual-snapshots-diff
         uses: getsentry/action-visual-snapshot@v2
-        # Do not execute on forks or on master. Forks are handled in visual-diff.yml
-        if: github.head_repository.full_name == 'getsentry/sentry' && github.ref != 'refs/heads/master'
+        # Forks & master are handled in visual-diff.yml
+        if: github.event.pull_request.head.repo.full_name == 'getsentry/sentry'
```
Comment (armenzg) [inline .github/workflows/acceptance.yml:328]: This is the actual fix. The changes on `visual-diff` is to duplicate the execution.

File: .github/workflows/visual-diff.yml (hunk 1 above)
Comment (asottile-sentry) [reply .github/workflows/visual-diff.yml:16]: just checking, if a branch in a fork is named `master` does it still do the right thing?

Comment (armenzg) [reply .github/workflows/visual-diff.yml:16]: > just checking, if a branch in a fork is named `master` does it still do the right thing?

Thanks for checking. We have never supported that. We only use workflow_run for PRs from Sentry forks. That will help us make sure their changes do not regress the acceptance tests.

**Review by asottile-sentry — APPROVED** [review]

**Comment by armenzg** [comment]
For PRs on upstream I can see the snapshot being triggered:
https://github.com/getsentry/sentry/runs/6194221741?check_suite_focus=true
image
and the workflow_run skipping (as expected):
https://github.com/getsentry/sentry/actions/runs/2233360532
image

For commits on master, I see `visual-diff` scheduling but complaining in [the log](https://github.com/getsentry/sentry/runs/6185746984?check_suite_focus=true#step:2:870) (which seems to be happening from before this merge):

```
##[debug]Push event triggered `workflow_run`... skipping as this only works for PRs
```

**Comment by armenzg** [comment]
A fork did [not execute](https://github.com/getsentry/sentry/runs/6195543591?check_suite_focus=true) via acceptance.yml but it did via visual-diff.yml:
image

**Pull request merged**